planning/obstacle_cruise_planner/src/planner_interface.cpp

```diff
@@ -260,12 +264,19 @@ double PlannerInterface::calcDistanceToCollisionPoint(
                           ? std::abs(vehicle_info_.max_longitudinal_offset_m)
                           : std::abs(vehicle_info_.min_longitudinal_offset_m);
 
+  const auto ego_segment_idx = motion_utils::findNearestSegmentIndex(
+    planner_data.traj.points, planner_data.current_pose, ego_nearest_param_.dist_threshold,
+    ego_nearest_param_.yaw_threshold);
```

Comment:
We have to make sure that ego_segment_idx is not none before it will be used.
Anyway we should use findFirstNearestSegmentIndex here.